File: 프로그래머스/lv2/42885. 구명보트/구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0;
    std::sort(people.begin(), people.end());
    int first = 0, last = people.size()-1;
    while(first <= last) {
        if (first == last) {
            answer++;
            break;
        }
        int min = people[first];
        int max = people[last];
        if (min+max <= limit) {
            first++;
            last--;
            answer++;
        } else {
            last--;
            answer++;
        }
    }
    
    return answer;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;
class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        int end = people.length -1;
        Arrays.sort(people);
        for (int i = 0; i <= end; i++) {
            if (i == end) {
                answer++;
                break;
            }
            if (people[i] + people[end] <= limit) {
                answer++;
                end--;
            } else {
                answer++;
                end--;
                i--;
            }
        }
        return answer;
    }
}